protocols/imap/src/main/java/org/apache/james/imap/api/ImapConfiguration.java

```diff
@@ -173,8 +181,9 @@ public ImapConfiguration build() {
     private final boolean isCondstoreEnable;
     private final boolean provisionDefaultMailboxes;
     private final Properties customProperties;
+    private final ImmutableSet<String> connectionChecks;
```

Comment:
```suggestion
    private final ImmutableSet<String> additionalConnectionChecks;
```

to distinguish with built-in connection check.

Shift+F6